react/features/whiteboard/functions.ts

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable import/order */
+import md5 from 'js-md5';
+
+// @ts-ignore
+import { getPinnedParticipant } from '../../features/base/participants';
+import { getRemoteParticipants, isLocalParticipantModerator } from '../base/participants/functions';
+import { appendURLParam } from '../base/util/uri';
+
+// @ts-ignore
+import { getCurrentRoomId, isInBreakoutRoom } from '../breakout-rooms/functions';
+
+// @ts-ignore
+import { _electParticipantInLargeVideo } from '../large-video/actions.any';
+import { hideWhiteboard, showWhiteboard, openWhiteboard } from './actions';
+import { WHITEBOARD_ID } from './constants';
+
+const getWhiteboardState = (state: any): any => state['features/whiteboard'];
+
+// TODO: remove negation
+/**
+ * Indicates whether the whiteboard is enabled in the config.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardEnabled = (state: any): boolean => !state['features/base/config'].whiteboard?.enabled;
+
+/**
+ * Indicates whether the whiteboard is open.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardOpen = (state: any): boolean => getWhiteboardState(state).isOpen;
+
+/**
+ * Indicates whether the whiteboard is present as a meeting participant.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardPresent = (state: any): boolean => getRemoteParticipants(state).has(WHITEBOARD_ID);
+
+/**
+ * Returns the whiteboard collaboration link.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {{ roomId: string, roomKey: string}|null}
+ */
+export const getCollabDetails = (state: any): {
+    roomId: string; roomKey: string;
+} => getWhiteboardState(state).collabDetails || {};
+
+/**
+ * Returns the whiteboard collaboration server url.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {string}
+ */
+export const getCollabServerUrl = (state: any): string => {
+    const collabServerBaseUrl = state['features/base/config'].whiteboard?.collabServerBaseUrl
+    || 'https://excalidraw-backend-pilot.jitsi.net'; // TODO: remove default
+
+    const { locationURL } = state['features/base/connection'];
+    const inBreakoutRoom = isInBreakoutRoom(state);
+    const roomId = getCurrentRoomId(state);
+    const room = md5.hex(`${locationURL?.href}${inBreakoutRoom ? `|${roomId}` : ''}`);
+
+    return appendURLParam(collabServerBaseUrl, 'room', room);
+};
+
+/**
+ * Whether the whiteboard is pinned.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardPinned = (state: any): boolean =>
+    getPinnedParticipant(state)?.id === WHITEBOARD_ID
+    || _electParticipantInLargeVideo(state) === WHITEBOARD_ID;
```

Comment:
This looks very odd. Why do you need need the 2nd part?

Reply:
Because the whiteboard might not be pinned while taking the stage, and I'm interested in knowing whether it's showing on stage at all, pinned or not, in order to set the Show/Hide option in the contextual menu and to change the display value in the whiteboard component. I should rename the selector, but I couldn't think of a good name.. isWhiteboardVisible?